### PR TITLE
include go routines

### PR DESCRIPTION
**Pull request by PriyaSudip**
Adding go routines to run queue agent querying in parallel. 

With agents:
image
Without queue:
image

### CODE REVIEW

File: internal/cluster/query.go [hunk 1]
```diff
@@ -27,18 +28,37 @@ func QueryCluster(ctx context.Context, OrganizationSlug string, ClusterID string
 		Queues:           []Queue{},
 	}
 
+	queueChannel := make(chan Queue, len(q.Organization.Cluster.Queues.Edges))
+	errorChannel := make(chan error, len(q.Organization.Cluster.Queues.Edges))
```
Comment (mcncl) [inline internal/cluster/query.go:32]: This error channel doesn't seem to ever be read from?

**Review by mcncl — CHANGES_REQUESTED** [review]
I wonder if we should be using more `context.WithTimeout` in our code to handle long API call issues.

File: internal/cluster/query.go (hunk 1 above)
Comment (PriyaSudip) [reply internal/cluster/query.go:32]: What would be the best place to send this information?

**Review by jradtilbrook — CHANGES_REQUESTED** [review]
This looks okay but its all done manually. This page https://gobyexample.com/waitgroups links out to the errgroup package which will make this much simpler: https://pkg.go.dev/golang.org/x/sync/errgroup

If you search the codebase, you'll see an example of it already in use as well I think

File: internal/cluster/query.go (hunk 1 above)
Comment (PriyaSudip) [reply internal/cluster/query.go:32]: Used errgroup to handle this

**Review by mcncl — APPROVED** [review]

**Comment by PriyaSudip** [comment]
> This looks okay but its all done manually. This page https://gobyexample.com/waitgroups links out to the errgroup package which will make this much simpler: https://pkg.go.dev/golang.org/x/sync/errgroup
> 
> If you search the codebase, you'll see an example of it already in use as well I think

Thanks @jradtilbrook as discussed I have made changes to include err group and instead of channel added array of queues and go routines based on the queue index

**Review by jradtilbrook — APPROVED** [review]
Sweet 👍

**Pull request merged**